### ISSUE TITLE
Prevent flashing background color of the preview when adding nodes to a curve using on-canvas picker

### ISSUE THREAD

**Issue by Beep6581** [Beep6581/RawTherapee#4743]
When adding nodes to a curve using the on-canvas pipette, the background color of the preview is temporarily changed to "theme-based", so if you use cropping and a background color other than "theme-based" the background distractingly flashes. Would be nice if it didn't.

See the screencast here:
https://discuss.pixls.us/t/kodak-q-13-color-target/8674/19?u=morgan_hardwood

**Comment by heckflosse** [comment]
Fix:

```diff
diff --git a/rtgui/cropwindow.cc b/rtgui/cropwindow.cc
index 77118277..3865591c 100644
--- a/rtgui/cropwindow.cc
+++ b/rtgui/cropwindow.cc
@@ -1379,7 +1379,7 @@ void CropWindow::expose (Cairo::RefPtr<Cairo::Context> cr)
                 break;
             }
         }
-        bool useBgColor = (state == SNormal);
+        bool useBgColor = (state == SNormal || state == SDragPicker || state == SDeletePicker);
     
         if (cropHandler.cropPixbuf) {
             imgW = cropHandler.cropPixbuf->get_width ();

```

**Comment by Beep6581** [comment]
@heckflosse
1. Open raw, apply neutral.
1. Draw a partial crop.
1. Zoom to fit whole image on screen (shortcut: alt+f).
1. Set background color of the preview to middle gray (shortcut: 9).
1. Now use any curve's picker button to place a node on the curve using the preview. The crop mask disappears while placing the curve.

**Comment by heckflosse** [comment]
@Beep6581 

```diff
diff --git a/rtgui/cropwindow.cc b/rtgui/cropwindow.cc
index c02f3632..040ccd01 100644
--- a/rtgui/cropwindow.cc
+++ b/rtgui/cropwindow.cc
@@ -1381,7 +1381,7 @@ void CropWindow::expose (Cairo::RefPtr<Cairo::Context> cr)
                 break;
             }
         }
-        bool useBgColor = (state == SNormal || state == SDragPicker || state == SDeletePicker);
+        bool useBgColor = (state == SNormal || state == SDragPicker || state == SDeletePicker || state == SEditDrag1);
     
         if (cropHandler.cropPixbuf) {
             imgW = cropHandler.cropPixbuf->get_width ();

```

**Comment by Beep6581** [comment]
@heckflosse sorry for the delay, fix confirmed.

**Comment by Thanatomanic** [comment]
Fixed with a7b7f20.
